src/builtins-decl.h

```diff
@@ -0,0 +1,928 @@
+/*
+  Copyright (c) 2024, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+// This file has to be the only place to define builtin symbol names that used
```

Comment:
I see two topics that you are trying to cover here:
- what builtins are;
- what the technical solution to deal with them in the source code is.

So to make the message more cleat I propose to structure it something like this (feel free to use this text or reformulate it for your taste): 

Builtins are target-specific functions implemented using LLVM IR. Their names typically start with double underscores. Unlike Standard Library functions, they are not guaranteed to be stable as compiler evolves. But they are used to implement the Standard Library.

During compilation flow ISPC compiler sometimes needs to treat builtins differently than user functions or stdlib functions, or do optimizations that recognize certain builtins (like gather/scatter optimization flow). Addressing builtins by string representation of the name in the source code is error prone, so this file solves this problem by defining the pointer with builtin name pointing to a C-string representing the name. For example, instead of using "__any" identifier, it's available as `builtin::__any`.

The list is alphabetically sorted for convenience.

Reply:
Thanks for reformulating it.